### DISCUSSION TITLE
Responses not as an embed

### DISCUSSION THREAD

**Discussion by warmbo** [jakobdylanc/llmcord.py#25] [Ideas]
It would be great if responses from the bot were not embeds and instead just plain messages similar to https://github.com/hc20k/LLMChat. The added benefits would include Discord's code formatting and a more comfortable interaction in the servers with users.

**Comment by jakobdylanc** [comment]
Thanks for the suggestion! I chose to use embeds because they have advantages like:
- Much higher character limit (4096) vs. regular messages (2000)
- A "color" parameter that I use as a helpful indicator for streaming status (orange=generating, green=done)

Also, embeds do support code formatting:
![image](https://github.com/jakobdylanc/discord-llm-chatbot/assets/38699060/8e88d019-3a59-4cce-abb2-077238dd654f)

Definitely a valid suggestion though. What are your thoughts?

**Reply by warmbo** [reply]
Your described benefits far outweigh my compulsive wants. :) I suppose my model was not responding correctly to format the code, using normal llama-7b I do now see that works! Thanks for this great tool. I've woo'd my friends with it and now have a favorite way of interacting with my local LLM.

**Reply by jakobdylanc** [reply]
Awesome to hear :) thanks for sharing, enjoy!

**Reply by warmbo** [reply]
I am curious about another topic, is it possible for the bot to be aware of the server in some way? Will the LLM be able to 'see' the server truly and tell me things like "there has been 20 posts in #channel today made by @person" ?

**Reply by jakobdylanc** [reply]
This is definitely possible, but a whole other beast and adds a lot of complexity.

You'd have to use something like chromadb (https://github.com/chroma-core/chroma) to maintain a vector database of messages for each channel, which you could then search for relevant messages to pull into the conversation. I've experimented with adding this but couldn't get to something I'm happy with.

I've also considered simply pulling in recent messages from whoever the user mentions in their message, so for example `@bot show me recent messages from @john` would pull recent messages from user `@john` into the conversation. This is on my to-do list to experiment with more.
